index.js

```diff
@@ -57,6 +73,34 @@ router
     ctx.status = 202
     ctx.body = 'OK'
   })
+  .get('/cloudflareError.png', async ctx => {
+    if (ctx.query['cloudflareErrorType'] == undefined || ctx.query['rayID'] == undefined || ctx.query['clientIP'] == undefined) {
+      ctx.status = 404
+    }
+    else if (!['500', '1000'].includes(ctx.query['cloudflareErrorType'])) {
+      console.error('Unregistered error type')
+      ctx.status = 404
+    }
+    else if (!rayIDRegex.test(ctx.query['rayID'])) {
```

Comment:
Re speed, is it really necessary to validate the ray ID value? If it is, can it reasonably be reduced to just checking that the length is 16 bytes?

Reply:
👍 sounds good